pyEpiabm/pyEpiabm/tests/test_sweep/test_spatial_sweep.py

```diff
@@ -52,31 +64,99 @@ def test__call__(self, mock_inf, mock_force, mock_poisson):
 
         # Change person's status to infected
-        self.infector.update_status(pe.property.InfectionStatus.InfectMild)
+        Parameters.instance().do_CovidSim = False
+        self.infector.update_status(InfectionStatus.InfectMild)
+        test_sweep.bind_population(self.pop)
+        test_sweep(time)
+        self.assertEqual(self.cell_susc.person_queue.qsize(), 1)
+
+        mock_dist.return_value = 0
         self.cell_susc.person_queue = Queue()
         test_sweep.bind_population(self.pop)
         test_sweep(time)
         self.assertEqual(self.cell_susc.person_queue.qsize(), 1)
 
-        # Add multiple people to infector cell
-        self.cell_inf.microcells[0].add_people(10000)
+        mock_dist.return_value = 2
+        Parameters.instance().do_CovidSim = True
         self.cell_susc.person_queue = Queue()
         test_sweep.bind_population(self.pop)
         test_sweep(time)
         self.assertEqual(self.cell_susc.person_queue.qsize(), 1)
 
-        # Change infectee's status to recovered
-        self.infectee.update_status(pe.property.InfectionStatus.Recovered)
+        # Change infectee's status to recovered so no susceptibles
+        Parameters.instance().do_CovidSim = False
+        self.infectee.update_status(InfectionStatus.Recovered)
         self.cell_susc.person_queue = Queue()
         test_sweep.bind_population(self.pop)
         test_sweep(time)
         self.assertTrue(self.cell_susc.person_queue.empty())
 
+        Parameters.instance().do_CovidSim = True
+        self.infectee.update_status(InfectionStatus.Recovered)
+        self.cell_susc.person_queue = Queue()
+        test_sweep.bind_population(self.pop)
+        test_sweep(time)
+        self.assertTrue(self.cell_susc.person_queue.empty())
+
+        # Add a cell with mostly recovered. Mock force ensure no
+        # infection events.
+        mock_force.return_value = 0.0
+        Parameters.instance().do_CovidSim = False
+        self.cell_susc.microcells[0].add_people(100)
+        self.cell_inf.persons[0].update_status(InfectionStatus.InfectMild)
+        for person in self.cell_susc.persons[:-1]:
+            person.update_status(InfectionStatus.Recovered)
+        self.cell_susc.person_queue = Queue()
+        test_sweep.bind_population(self.pop)
+        test_sweep(time)
+        self.assertTrue(self.cell_susc.person_queue.empty())
+
+        # Add third cell to cover different distances
+        mock_dist.side_effect = [0, 2]
+        self.pop.add_cells(1)
+        self.third_cell = self.pop.cells[2]
+        self.third_cell.add_microcells(1)
+        self.third_cell.microcells[0].add_people(1)
+        self.cell_susc.person_queue = Queue()
+        test_sweep.bind_population(self.pop)
+        test_sweep(time)
+        enqueued_people = (self.cell_susc.person_queue.qsize()
+                           + self.third_cell.person_queue.qsize())
+        self.assertEqual(enqueued_people, 0)
```

Comment:
Maybe we should test that this is giving the weightings we actually expect?

Reply:
Finding it really hard to test these explicitly, as the only thing that can really be tracked is how the queue size change, and excessive amount will obviously not help that.

Reply:
Mocked up the np.nan_to_num function and checked it was called in the right places/right arguments